Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,258 @@
+
+            return biggestLoss;
+        }
+
+        private decimal GetBiggestLoss(decimal biggestLoss, decimal beginAmount, decimal endAmount)
+        {
+            var loss = endAmount - beginAmount;
+            var bigLoss = biggestLoss;
+
+            if (loss < bigLoss)
+            {
+                bigLoss = loss;
+            }
+
+            return bigLoss;
+        }
+
+        /// <summary>
+        /// Return name and current money of the richest person.
+        /// </summary>
+        public string FindRichestPerson(string[] peopleAndBalances)
+        {
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return NotAvailable;
+            }
+
+            var maxBalance = 0m;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid)
+                {
+                    SetNames(ref name, ref maxBalance, peopleBalances, balance);
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word1 = name.ToString().Contains(", ") ? "are" : "is";
+            var word2 = name.ToString().Contains(", ") ? "people" : "person";
+            return $"{FixPlural(name.ToString())} {word1} the richest {word2}. {maxBalanceNoComma}.";
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public string FindMostPoorPerson(string[] peopleAndBalances)
+        {
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return NotAvailable;
+            }
+
+            var minBalance = decimal.MaxValue;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid)
+                {
+                    GetPoorest(ref name, ref minBalance, peopleBalances, balance);
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var minBalanceNoComma = RemoveComma($"{minBalance:C0}").Replace("(", "-").Replace(")", "");
+            var word = name.ToString().Contains(", ") ? "have" : "has";
+            return $"{FixPlural(name.ToString())} {word} the least money. {minBalanceNoComma}.";
+        }
+
+        private void GetPoorest(ref StringBuilder mostPoorPerson, ref decimal minBalance, string[] peopleBalances, decimal balance)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetPoorest' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)